src/axom/inlet/tests/inlet_Reader.cpp

```diff
@@ -132,36 +151,54 @@ TYPED_TEST(inlet_Reader, getMap)
   reader.parseString(fromLuaTo<TypeParam>(testString));
 
   std::unordered_map<int, int> ints;
-  bool found = reader.getIntMap("luaArray", ints);
-  EXPECT_TRUE(found);
+  ReaderResult retValue = reader.getIntMap("luaArray", ints);
+  EXPECT_EQ(retValue, ReaderResult::NotHomogeneous);
   std::unordered_map<int, int> expectedInts {{0, 4}, {1, 5}, {2, 6}, {5, 2}};
   EXPECT_EQ(expectedInts, ints);
 
   std::unordered_map<int, double> doubles;
-  found = reader.getDoubleMap("luaArray", doubles);
-  EXPECT_TRUE(found);
+  retValue = reader.getDoubleMap("luaArray", doubles);
+  EXPECT_EQ(retValue, ReaderResult::NotHomogeneous);
```

Comment:
How does casting work?  And are there tests for `ReaderResult::NotHomogenous` specifically related to casting?

E.g. if the `testString` above only contained `int` and `double` types, and you called
* `reader.getDoubleMap("luaArray", doubles);`
* `reader.getIntMap("luaArray", ints);
I'd imagine the former would cast the ints to doubles and return Success, but the latter would return `NotHomogenous`

(apologies if this is already there, it can be a bit difficult to see it via the github interface)

Reply:
This is a good question - to provide a little background on integral <-> floating-pt promotion/demotion in inlet, most of the implementation is restricted by Lua's decision to use the same type for integral and floating-point values.  To maintain consistency, I implemented special promotion/demotion logic in the YAML/JSON readers so they'd behave like the Lua reader for consistency.

As such, Inlet doesn't provide a way to see if an integral <-> floating-pt conversion occurred and therefore casting doesn't (can't?) affect the homogeneity of an array.